gtsfm/evaluation/metrics.py

```diff
@@ -0,0 +1,399 @@
+                      Has to be provided if data = None.
+             store_full_data: Whether all the values have to be stored and saved or only summary is required. True by default.
+             plot_type: The plot to use for visualization of the metric.
+                        Defaults:
+                           PlotType.BAR if data is a scalar
+                           PlotType.BOX if data is a distribution (other option is PlotType.HISTOGRAM)
+                         It is inferred from the summary if plot_type is not provided and summary is.
+        """
+        if summary is None and data is None:
+            raise ValueError("Data and summary cannot both be None.")
+
+        self._name = name
+        if data is not None:
+            # Cast to a numpy array
+            if not isinstance(data, np.ndarray):
+                data = np.array(data)
+            if data.ndim > 1:
+                raise ValueError("Metrics must be scalars on 1D-distributions.")
+
+            # Save dimension and plot_type for data
+            self._dim = data.ndim
+            plot_types_for_dim = self._get_plot_types_for_dim(self._dim)
+            if plot_type is None:
+                if summary is not None:
+                    self._plot_type = self.PlotType.HISTOGRAM if "histogram" in summary else self.PlotType.BOX
+                else:
+                    self._plot_type = plot_types_for_dim[0]
+            elif plot_type in plot_types_for_dim:
+                self._plot_type = plot_type
+            else:
+                raise ValueError("Unsupported plot type for the data dimension")
+
+            # Create a summary if the data is a 1D distribution
+            if self._dim == 1:
+                self._summary = self._create_summary(data)
+
+            # Store full data only if its a scalar or if asked to.
+            if self._dim == 0 or store_full_data:
+                self._data = data
+            else:
+                self._data = None
+        else:
+            # Metrics created from summary alone are 1D distribution metrics
+            self._dim = 1
+            self._summary = summary
+            self._plot_type = self.PlotType.HISTOGRAM if "histogram" in summary else self.PlotType.BOX
+            self._data = None
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @property
+    def data(self) -> np.ndarray:
+        return self._data
+
+    @property
+    def plot_type(self) -> PlotType:
+        return self._plot_type
+
+    @property
+    def dim(self) -> int:
+        return self._dim
+
+    @property
+    def summary(self) -> Dict[str, Any]:
+        return self._summary
+
+    def _get_plot_types_for_dim(self, dim: int) -> List[PlotType]:
+        if dim == 0:
+            return [self.PlotType.BAR]
+        if dim == 1:
+            return [self.PlotType.BOX, self.PlotType.HISTOGRAM]
+        return []
```

Comment:
is a value error better, if dim is not 0 or 1?

Reply:
I don't think it is an error, if it's not a supported dimension, it does not return any plot types. That is expected behavior.